src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Webservice/WebserviceKeyType.php

```diff
@@ -0,0 +1,164 @@
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\Webservice;
+
+use PrestaShop\PrestaShop\Core\Domain\Webservice\ValueObject\Key;
+use PrestaShopBundle\Form\Admin\Type\GeneratableTextType;
+use PrestaShopBundle\Form\Admin\Type\Material\MaterialMultipleChoiceTableType;
+use PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\CallbackTransformer;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Is used to create form for adding/editing Webservice Key
+ */
+class WebserviceKeyType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var bool
+     */
+    private $isMultistoreFeatureUsed;
+
+    /**
+     * @var array
+     */
+    private $resourceChoices;
+
+    /**
+     * @var array
+     */
+    private $permissionChoices;
+
+    /**
+     * @param bool $isMultistoreFeatureUsed
+     * @param array $resourceChoices
+     * @param array $permissionChoices
+     */
+    public function __construct(
+        $isMultistoreFeatureUsed,
+        array $resourceChoices,
+        array $permissionChoices
+    ) {
+        $this->isMultistoreFeatureUsed = $isMultistoreFeatureUsed;
+        $this->resourceChoices = $resourceChoices;
+        $this->permissionChoices = $permissionChoices;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('key', GeneratableTextType::class, [
+                'generated_value_length' => Key::LENGTH,
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
```

Comment:
_This field is required_? Otherwise, _The %s field is required._ is already in stock.

Reply:
done.